### PR TITLE
feat: pio/v0.44.5

### PR DESCRIPTION
**Pull request by fkneeland-figure**
## Description

Closes: #XXXX

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: RELEASE_NOTES.md [hunk 1]
```diff
@@ -1,10 +1,3 @@
...
 
-This release introduces bug fixes and improvements on the Cosmos SDK v0.44 series:
-
-- Emit ante handler events for failed transactions: ant events can cause blockchain change (eg tx fees) and related events should be emitted.
-- (fix) Upgrade IAVL to 0.17.3 to solve race condition bug in IAVL.
-
-See the [Cosmos SDK v0.44.5 Changelog](https://github.com/cosmos/cosmos-sdk/blob/v0.44.5/CHANGELOG.md) for the exhaustive list of all changes.
-
-**Full Changelog**: https://github.com/cosmos/cosmos-sdk/compare/v0.44.4...v0.44.5
+This release adds the fixes from cosmos-sdk upstream v0.44.4 and v0.44.5 to the updated Provenanced version v0.44.3
```
Comment (fkneeland-figure) [inline RELEASE_NOTES.md:3]: TODO: Need to update the wording on this.

File: CHANGELOG.md [hunk 2]
```diff
@@ -60,6 +61,14 @@ Ref: https://keepachangelog.com/en/1.0.0/
 * (bank) [\#10394](https://github.com/cosmos/cosmos-sdk/pull/10394) Fix: query account balance by ibc denom.
 * [\10608](https://github.com/cosmos/cosmos-sdk/pull/10608) Change the order of module migration by pushing x/auth to the end. Auth module depends on other modules and should be run last. We have updated the documentation to provide more details how to change module migration order. This is technically a breaking change, but only impacts updates between the upgrades with version change, hence migrating from the previous patch release doesn't cause new migration and doesn't break the state.
 
+
+## [v0.44.3-rosetta-fix](https://github.com/provenance-io/cosmos-sdk/tag/v0.44.3-rosetta-fix)
+
+### Improvements
```
Comment (arnabmitra) [inline CHANGELOG.md:67]: ugh..we forgot to add the msg fee Improvement here, i will work on adding it :(

File: baseapp/abci.go [hunk 3]
```diff
@@ -240,7 +240,7 @@ func (app *BaseApp) CheckTx(req abci.RequestCheckTx) abci.ResponseCheckTx {
 		panic(fmt.Sprintf("unknown RequestCheckTx type: %s", req.Type))
 	}
 
-	gInfo, result, anteEvents, err := app.runTx(mode, req.Tx)
+	gInfo, result, anteEvents, _, err := app.runTx(mode, req.Tx)
```
Comment (arnabmitra) [inline baseapp/abci.go:243]: needs a linter run i think

File: contrib/images/simd-env/Dockerfile [hunk 4]
```diff
@@ -2,7 +2,6 @@ FROM golang:1.17-alpine AS build
 RUN apk add build-base git linux-headers
 WORKDIR /work
 COPY go.mod go.sum /work/
-COPY db/go.mod db/go.sum /work/db/
```
Comment (fkneeland-figure) [inline contrib/images/simd-env/Dockerfile:5]: This was deleted as the db/ folder does not exist in cosmos-sdk `v0.44.5` and thus causes a `make localnet-start` to fail as well as failing tests.

File: RELEASE_NOTES.md [hunk 5]
```diff
@@ -1,10 +1,3 @@
...
 
-This release introduces bug fixes and improvements on the Cosmos SDK v0.44 series:
-
-- Emit ante handler events for failed transactions: ant events can cause blockchain change (eg tx fees) and related events should be emitted.
-- (fix) Upgrade IAVL to 0.17.3 to solve race condition bug in IAVL.
-
-See the [Cosmos SDK v0.44.5 Changelog](https://github.com/cosmos/cosmos-sdk/blob/v0.44.5/CHANGELOG.md) for the exhaustive list of all changes.
-
-**Full Changelog**: https://github.com/cosmos/cosmos-sdk/compare/v0.44.4...v0.44.5
+This release adds the updates from cosmos-sdk upstream v0.44.4 and v0.44.5 to the msg fee and rosetta fixes added into the Provenanced version v0.44.3
```
Comment (channa-figure) [inline RELEASE_NOTES.md:3]: NIT: Can we say provenance fork instead of provenanced version?

File: baseapp/baseapp.go [hunk 6]
```diff
@@ -133,6 +133,9 @@ type BaseApp struct { // nolint: maligned
 	// indexEvents defines the set of events in the form {eventType}.{attributeKey},
 	// which informs Tendermint what to index. If empty, all events will be indexed.
 	indexEvents map[string]struct{}
+
+	// fee handler
```
Comment (channa-figure) [inline baseapp/baseapp.go:137]: NIT: Could probably remove this comment.  The variable name says the same thing.

**Review by channa-figure — CHANGES_REQUESTED** [review]

File: baseapp/baseapp.go (hunk 6 above)
Comment (arnabmitra) [reply baseapp/baseapp.go:137]: i'll take it

File: baseapp/baseapp.go [hunk 7]
```diff
@@ -678,15 +686,35 @@ func (app *BaseApp) runTx(mode runTxMode, txBytes []byte) (gInfo sdk.GasInfo, re
 	// Result if any single message fails or does not have a registered Handler.
 	result, err = app.runMsgs(runMsgCtx, msgs, mode)
 	if err == nil && mode == runTxModeDeliver {
-		msCache.Write()
-
-		if len(anteEvents) > 0 {
-			// append the events in the order of occurrence
-			result.Events = append(anteEvents, result.Events...)
```
Comment (arnabmitra) [inline baseapp/baseapp.go:685]: we definitely need the anteEvents added here?

Comment (arnabmitra) [reply baseapp/baseapp.go:685]: can someone re-review the ante handler  events being propagated correctly.

Comment (fkneeland-figure) [reply baseapp/baseapp.go:685]: @channa-figure Could you?

**Review by channa-figure — APPROVED** [review]
LGTM

**Review by arnabmitra — APPROVED** [review]
lgtm

**Pull request merged**